Public/Src/Cache/MemoizationStore/Interfaces/Sessions/ContentHashList.cs

```diff
@@ -67,6 +68,36 @@ public class ContentHashList : IEquatable<ContentHashList>
             _payload = payload;
         }
 
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="ContentHashList" /> class.
+        /// </summary>
+        public ContentHashList(BinaryReader reader)
+        {
+            Contract.Requires(reader != null);
+            var numHashes = reader.ReadInt32();
+            if (numHashes < 0)
+            {
+                _contentHashes = null;
+            }
+            else
+            {
+                _contentHashes = new ContentHash[numHashes];

```

Comment:
>_contentHashes [](start = 16, length = 14)

It seems that before your change, _contentHashes was a non-nullable field, but now it can be null.

I think if numHashes is < 0, then _contentHashes should be an empty array:

_contentHashes = CollectionUtils.EmptyArray<T>(); (or something).

Reply:
Great catch! numHashes can never be <0 because it is -1 if _contentHashes == null and the length of the array otherwise. Since the first thing can never happen, it is always the length of an array. Changed accordingly.

---
In reply to: [302181279](https://github.com/microsoft/BuildXL/pull/571#discussion_r302181279) [](ancestors = 302181279)